src/mono/CMakeLists.txt

```diff
@@ -183,11 +187,17 @@ endif()
 # HOST OS CHECKS
 ######################################
 
+if("${triple}" STREQUAL "wasm32-wasi-threads")
+  set(CMAKE_SYSTEM_VARIANT "threads")
+endif()
+
 message ("CMAKE_SYSTEM_NAME=${CMAKE_SYSTEM_NAME}")
 message ("CMAKE_SYSTEM_VARIANT=${CMAKE_SYSTEM_VARIANT}")
 
 set(CLR_CMAKE_HOST_OS ${CMAKE_SYSTEM_NAME})
-string(TOLOWER ${CLR_CMAKE_HOST_OS} CLR_CMAKE_HOST_OS)
+set(CLR_CMAKE_HOST_VARIANT ${CMAKE_SYSTEM_VARIANT})
```

Comment:
i'm not super familiar with cmake, but it seems weird that the *host* variant is either threads or not threads depending on the target you're building for. don't all cmake build hosts have threads?

Reply:
I think this is referring not to build host but mono runtime host. (as opposed to target system for which runtime will do codegen)
Still, I'm not sure how that interacting with `DISABLE_THREADS` (shouldn't there be TARGET/HOST_DISABLE_THREADS?), but in any case it was used for wasm this way before.